### PR TITLE
refactor: noise cancellation component, add to mweb when supported

### PR DESCRIPTION
**Pull request by amar-1995**

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx [hunk 1]
```diff
@@ -264,7 +272,14 @@ export const PreviewControls = ({ hideSettings, vbEnabled }: { hideSettings: boo
         <AudioVideoToggle />
         {vbEnabled ? <VBToggle /> : null}
       </Flex>
-      {!hideSettings ? <PreviewSettings /> : null}
+      <Flex align="center" gap="1">
+        {isMobile && isNoiseCancellationEnabled && isKrispPluginAdded ? (
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx:276]: it can be shown otherwise also no? why are we showing only when added? if it is not enabled by default, they should be able to enable.

Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx:276]: we should krispPlugin.checkSupport.isSupported as welll.

File: packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx [hunk 2]
```diff
@@ -197,6 +209,19 @@ export const MwebOptions = ({
                 <ActionTile.Title>{isHandRaised ? 'Lower' : 'Raise'} Hand</ActionTile.Title>
               </ActionTile.Root>
             ) : null}
+            {krispPlugin.isSupported() && room.isNoiseCancellationEnabled && localPeerAudioTrackID ? (
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx:212]: why is this needed? we can;t use the noisecancellation from AudioVideoToggle? it does have isMobile check no?

File: packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx (hunk 1 above)
Comment (amar-1995) [reply packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx:276]: This is only an icon in the preview screen that says whether noise cancellation is enabled.

File: packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx (hunk 2 above)
Comment (amar-1995) [reply packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx:212]: We need to show it more options in mweb

**Review by raviteja83 — APPROVED** [review]
